### ISSUE TITLE
[seo-magic] Exception / Error 500 when content item has no metadata:description and no default is set for the site.

### ISSUE THREAD

**Issue by k8n** [getgrav/grav-premium-issues#56]
"Undefined index: description"

```
/app/user/plugins/seo-magic/classes/SEOMagic.php, Line 171:
    $metadata['og:description'] = $header->{'seo-magic'}['opengraph']['description'] ?? $metadata['description'];
```

Context:
```
/app/user/plugins/seo-magic/classes/SEOMagic.php
            $robots_txt = $this->getRobotsFile();
 
            // Robots file doesn't exist, add robots meta even if it doesn't exist on page
            if (!$robots_txt) {
                $metadata['robots'] = $robots_meta;
            } elseif ($robots_meta_page) {
                $metadata['robots'] = $robots_meta_page;
            }
 
            if ($this->config->get('plugins.seo-magic.facebook.enabled')) {
                $metadata['fb:app_id'] = $this->config->get('plugins.seo-magic.facebook.app_id');
            }
 
            if ($this->config->get('plugins.seo-magic.opengraph') === true ||
                $this->config->get('plugins.seo-magic.opengraph.enabled')) {
                $metadata['og:url'] = $page->url(true);
                $metadata['og:type'] = 'website';
                $metadata['og:site_name'] = $this->config->get('site.title');
                $metadata['og:title'] = $header->{'seo-magic'}['opengraph']['title'] ?? $metadata['title'];
                $metadata['og:description'] = $header->{'seo-magic'}['opengraph']['description'] ?? $metadata['description'];
 
                if (isset($metadata['image'])) {
                    $metadata['og:image'] = $metadata['image'];
                }
            }
            if ($this->config->get('plugins.seo-magic.twitter.enabled')) {
                $metadata['twitter:card'] = $this->config->get('plugins.seo-magic.twitter.card');
                $metadata['twitter:site'] = $this->config->get('plugins.seo-magic.twitter.username');
                $metadata['twitter:title'] = $header->{'seo-magic'}['twitter']['title'] ?? $metadata['og:title'] ?? $metadata['title'];
                $metadata['twitter:description'] = $header->{'seo-magic'}['twitter']['description'] ?? $metadata['og:description'] ?? $metadata['description'];
 
                if (isset($metadata['image'])) {
                    $metadata['twitter:image'] = $metadata['image'];
                    $metadata['twitter:image:alt'] = $metadata['title'];
                }
                
                if ($this->config->get('plugins.seo-magic.twitter.reading_time', true) &&
                    $this->config->get('plugins.readingtime.enabled'))  {
                    $reading_time = new TwigReadingTimeFilters();
                    $metadata['twitter:label1'] = 'Est. reading time';

Arguments
"Undefined index: description"

```

**Comment by rhukster** [comment]
If you set a default in configuration -> site the problem goes away? 

It looks like we need an extra check if no default is set.

**Comment by rhukster** [comment]
Fixed for upcoming version 2.0

**Comment by rhukster** [comment]
released
